lib/msf/base/sessions/command_shell.rb

```diff
@@ -421,52 +421,68 @@ def cmd_download(*args)
 
     # Check if src exists
     if !_file_transfer.file_exist?(src)
-      print_error("The target file does not exist")
+      print_error('The target file does not exist')
       return
     end
 
+    if (::File.basename(dst) != File.basename(src))
+      # The destination when downloading is a local file so use this system's separator
+      dst += ::File::SEPARATOR + File.basename(src)
```

Comment:
No actually we don't because the target may not be the same platform as what we're running on. I'm not sure if shell sessions define File like Meterpreter sessions do but that's used to ensure that the correct separator is used among other things. Local paths should use Ruby's ::File while the remote path should use whatever is appropriate based on the platform.

Reply:
Ah okay thanks for clarifying that.